subsys/bluetooth/host/l2cap.c

```diff
@@ -451,15 +461,51 @@ static int l2cap_le_conn_req(struct bt_l2cap_le_chan *ch)
 	req->mps = sys_cpu_to_le16(ch->rx.mps);
 	req->credits = sys_cpu_to_le16(ch->rx.init_credits);
 
-	l2cap_chan_send_req(ch, buf, L2CAP_CONN_TIMEOUT);
+	l2cap_chan_send_req(&ch->chan, buf, L2CAP_CONN_TIMEOUT);
+
+	return 0;
+}
+
+static int l2cap_ecred_conn_req(struct bt_l2cap_chan **chan, int channels)
+{
+	struct net_buf *buf;
+	struct bt_l2cap_ecred_conn_req *req;
+	int i;
+	u8_t ident;
+
+	ident = get_ident();
+
+	buf = l2cap_create_le_sig_pdu(NULL, BT_L2CAP_ECRED_CONN_REQ, ident,
+				      sizeof(*req) +
+				      (channels * sizeof(u16_t)));
+
+	req = net_buf_add(buf, sizeof(*req));
+
+	for (i = 0; i < channels; i++) {
+		struct bt_l2cap_le_chan *ch = BT_L2CAP_LE_CHAN(chan[i]);
+		ch->chan.ident = ident;
+
+		/* Init common parameters */
+		if (!i) {
+			req->psm = sys_cpu_to_le16(ch->chan.psm);
+			req->mtu = sys_cpu_to_le16(ch->rx.mtu);
+			req->mps = sys_cpu_to_le16(ch->rx.mps);
+			req->credits = sys_cpu_to_le16(ch->rx.init_credits);
+		}
```

Comment:
You could consider moving this before the for-loop so you don't need to have a branch

Reply:
Will do.